### PR TITLE
Add Tedlium v3 recipe

### PR DESCRIPTION
**Pull request by Masao-Someki**

### CODE REVIEW

**Comment by kan-bayashi** [comment]
Please merge current v.0.4.0 changes and solve conflict.

```bash
git fetch upstream
git merge upstream/v.0.4.0
```

**Comment by Masao-Someki** [comment]
merged upstream/v.0.4.0

**Comment by sw005320** [comment]
Thanks for the update. I have questions/requests:
1. Could you use config files by following the latest recipes? See https://github.com/espnet/espnet/blob/v.0.4.0/egs/tedlium/asr1/run.sh
2. Compared with the Kaldi result https://github.com/kaldi-asr/kaldi/blob/master/egs/tedlium/s5_r3/RESULTS, the numbers seem to be bad (it's around the GMM performance.). Do you know what's happened?

**Comment by Masao-Someki** [comment]
1. sure. I will fix run.sh to use config files!
2. I'm sorry, I don't know what was the problem...
The acc.png and the loss.png are the following, so I think there's no problem with learning process.
* acc.png
image
* loss.png
image

for now, all the parametors are completely the same as tedlium-release2 recipe, so I will try some parametor sets for decoding process.

**Comment by sw005320** [comment]
Can you tell me the difference between tedlium2 and tedlium3?

**Comment by Masao-Someki** [comment]
I can't know what information you need, so I just mention about the difference in dataset.
The main difference between v2 and v3 is their size of dataset.  The following is the contents of v2 and v3.
```
v2 Contents:
– 1495 audio talks in NIST sphere format (SPH)
– 1495 transcripts in STM format
– Dictionary with pronunciation (159848 entries)
– Selected monolingual data for language modeling from WMT12 publicly available corpora

v3 Contents:
– 2351 audio talks in NIST sphere format (SPH), including talks from TED-LIUM 2: be careful, same talks but not same audio files (only these audio file must be used with the TED-LIUM 3 STM files)
–> 452 hours of audio
– 2351 aligned automatic transcripts in STM format

– TEDLIUM 2 dev and test data: 19 TED talks in SPH format with corresponding manual transcriptions (cf. ‘legacy’ distribution below).

– Dictionary with pronunciations (159848 entries), same file as the one included in TED-LIUM 2
– Selected monolingual data for language modeling from WMT12 publicly available corpora: these files come from the TED-LIUM 2 release, but have been modified to get a tokenization more relevant for English language
```
according to https://lium.univ-lemans.fr/ted-lium2/ and https://lium.univ-lemans.fr/ted-lium3/

**Comment by Masao-Someki** [comment]
Sorry about the above, I used the speaker-adaptation dataset. To compare with the result of v2, I  should have used the legacy set...
Now I run this script with the datasets, so wait a little until it finishes.

**Comment by kan-bayashi** [comment]
Is there any progress?
We should merge this PR before #793.

**Comment by Masao-Someki** [comment]
sorry to be late!
updated the result file with legacy datasets.

**Comment by sw005320** [comment]
OK. I don't know why the performance is still behind the tedlium v2, but we can think of it in a later PR.

- [ ] we should change the directory name `tedlium-v3` --> `tedlium3`, and the original one from `tedlium` --> `tedlium2`
- [ ] please keep the speaker adapation result in `RESULTS`

We can also think of the following in a later PR.
- [ ] use `RESULTS.md` instead of `RESULTS`. See https://github.com/espnet/espnet/blob/v.0.4.0/egs/tedlium/asr1/RESULTS.md. To make `RESULTS.md`, please use https://github.com/espnet/espnet/blob/v.0.4.0/utils/get_sys_info.sh and https://github.com/espnet/espnet/blob/v.0.4.0/utils/pack_model.sh

File: egs/tedlium3/asr1/RESULTS.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  - date: `Tue Jun 11 11:22:28 JST 2019`
+  - system information: `Linux chikaku1.sp.m.is.nagoya-u.ac.jp 3.10.0-862.14.4.el7.x86_64 #1 SMP Wed Sep 26 15:12:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
+  - python version: `Python 3.6.5`
+  - espnet version: `espnet 0.3.1`
+  - chainer version: `chainer 5.0.0`
+  - pytorch version: `pytorch 1.0.0`
+  - Git hash: `2679f69a46f5cfa103325ceffb53793d6af4af2b`
+
+- Model files (archived to model.tar.gz by `$ pack_model.sh`)
+    - model link: (put the model link manually. please contact Shinji Watanabe <shinjiw@ieee.org> if you want a web storage to put your files)
```
Comment (sw005320) [inline egs/tedlium3/asr1/RESULTS.md]: Could you upload your model to the Google drive?
I sent you an invitation.

File: egs/tedlium3/asr1/run.sh [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=train_trim_sp
+train_dev=dev_trim
+recog_set="dev test"
```
Comment (sw005320) [inline egs/tedlium3/asr1/run.sh]: Is it possible to keep both adaptation and legacy data directories, and just switch either option by setting the `train_set`, `train_dev`, and `recog_set` with corresponding comments? 
We could create both data directories in `local/prepare_data.sh`.
We can keep the current name for the legacy one, and name the adaptation directories as `train_adapt_trim_sp`, `dev_adapt_trim`, `dev_adapt`, and `test_adapt`

**Comment by sw005320** [comment]
@takenori-y, the recipe directory name would be changed from tedlium to tedlium2. I think it would not affect your work, but it's just a note.

File: egs/tedlium3/asr1/run.sh (hunk 2 above)
Comment (sw005320) [reply egs/tedlium3/asr1/run.sh]: For example,

```
# legacy setup
train_set=train_trim_sp
train_dev=dev_trim
recog_set="dev test"
# adaptation setup
#train_set=train_adapt_trim_sp
#train_dev=dev_adapt_trim
#recog_set="dev_adapt test_adapt"
```
and generate the both legacy and adaptation direcotries in `local/prepare_data.sh`.

File: egs/tedlium3/asr1/RESULTS.md (hunk 1 above)
Comment (Masao-Someki) [reply egs/tedlium3/asr1/RESULTS.md]: Sure. Thank you.

File: egs/tedlium3/asr1/run.sh (hunk 2 above)
Comment (Masao-Someki) [reply egs/tedlium3/asr1/run.sh]: Okay.

**Pull request merged**